homeassistant/components/emoncms/sensor.py

```diff
@@ -107,6 +106,10 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
         if sensor_names is not None:
             name = sensor_names.get(int(elem["id"]), None)
 
+        unit_of_measurement = elem.get("unit", "")
+        if unit_of_measurement == "":
+            unit_of_measurement = config.get(CONF_UNIT_OF_MEASUREMENT)
```

Comment:
Since it is already set in line 77, we do not need to fetch it again. Instead we only override the fetched value if the api has returned something:
```suggestion
            unit_of_measurement = unit
```

Reply:
No, it needs to do it again as the variable `unit_of_measurment` at this point is the last used value not the default unit.

Reply:
Yeah, you are right 👍 but then we need to change it a bit to:
```python
if unit:
  unit_of_measurement = unit
else:
  unit_of_measurement = config.get(CONF_UNIT_OF_MEASUREMENT)
```

Reply:
Ok if you prefer that to just setting `unit_of_measurment` twice.

Reply:
Probably easiest if you change and I'll accept!